spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,134 @@
 require 'rails_helper'
 
-RSpec.describe ProgramSession, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe ProgramSession do
+
+  let(:proposal) { create(:proposal_with_track, :with_two_speakers) }
+
+  it "responds to .create_from_proposal" do
+    expect(ProgramSession).to respond_to(:create_from_proposal)
+  end
+
+  describe "#create_from_proposal" do
+
+    it "creates a new program session with the same title as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.title).to eq(proposal.title)
+    end
+
+    it "creates a new program session with the same abstract as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.abstract).to eq(proposal.abstract)
+    end
+
+    it "creates a new program session with the same event as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.event_id).to eq(proposal.event_id)
+    end
+
+    it "creates a new program session with the same session format as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.session_format_id).to eq(proposal.session_format_id)
+    end
+
+    it "creates a new program session with the same track as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.track_id).to eq(proposal.track_id)
+    end
+
+    it "creates a program session that has the proposal id" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.proposal_id).to eq(proposal.id)
+    end
+
+    it "creates a program session that is active" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.state).to eq("active")
+    end
+
+    it "sets program session id for all speakers" do
```

Comment:
Funny story: if the program session id *didn't* get set, then the session's speakers association would be empty, that loop would get skipped over and the test would pass :D

Another approach would be to compare the contents of the session's speakers collection to with the proposal's speakers collection. `contain_exactly` or `match_array` might work: https://www.relishapp.com/rspec/rspec-expectations/v/3-5/docs/built-in-matchers/contain-exactly-matcher